rfcs/999-shipperctl-admin.adoc

```diff
@@ -0,0 +1,94 @@
+  context: gke-us-management # <2>
+
+applicationClusters:
+- name:   eu-application     # <1>
+  region: eu-west1
+  zone:   a
+  capabilities:
+    - gdpr
+    - ipv4
+    - kubernetes1.12
+- name:    us-application
+  context: gke-us-application # <2>
+  region:  us-west1
+  zone:    a
+  capabilities:
+    - ipv4
+    - ssd-local-disk
+----
+<1> Without the `context` field present, `name` is considered a valid and existing _kubectl_ context, and will be used both to gather connection details to perform relevant operations on the cluster to set up Shipper and to name cluster specific manifests in the Management Cluster.
+<2> With the `context` field present, the `name` value is used only to name cluster specific manifests in the Management Cluster. Connection details are gathered using the specified `context` value.
+
+.*shipperctl admin clusters apply* arguments
+|===
+|Option|Description
+
+|`-f, --file`
+| Shipper cluster configuration file to be applied
+
+|`--dry-run`
+| Print out what actions it would do, according to the given cluster configuration file
+
+|`-n, --namespace`
+| Shipper system namespace to be created in all clusters
+
+|`-k, --kubeconfig`
+| Same as _kubectl_
+
+|===
+
+There are two use cases for this command:
+
+* One use case is to set up a collection of management and application clusters for the first time. In this case, all of the operations required to set up a management/application cluster are done.
+* The second use case is to update the Cluster objects, or to re-do any of the steps that are necessary to add this cluster. For example, if a service account is already set up for Shipper, this command will skip that step. This makes this command idempotent, and allows it to be used for git-ops or any sort of pipeline.
+
+This command should do the following steps for each cluster:
+
+=== Set Up the Management Cluster
+
+* Create the Namespace based on the `namespace` argument on the management cluster
+* Register CRDs on the management cluster
+* Create a service account called `shipper` in the Shipper namespace on the management cluster
+* Create a ClusterRole called `shipper-admin` to manage Shipper resources on the management cluster
+* Create a ClusterRoleBinding to bind the shipper service account to the shipper-admin role on the management cluster
+* Create a Role called `secret-reader` in the Shipper namespace that gives permission to read secrets on the management cluster
+* Create a RoleBinding to bind the Shipper service account to the `secret-reader` role
+
+=== Set Up the Application Cluster
+
+* Create the Namespace based on the `namespace` argument on the application cluster
+* Create a service account called `shipper` in the namespace on the application cluster
+* Create a ClusterRoleBinding to bind the Shipper service account to the `cluster-admin` ClusterRole on the application cluster
+
+=== Connect the Clusters Together
+
+* Create the cluster Secret on the management cluster
```

Comment:
Just to add a missing bit: the contents of the Secret created on the Management cluster for this Application cluster should be the Service Account Token (which will be a secret called 'shipper-token-$hash' in the shipper namespace). This typically includes a few fields: `ca.crt`, `namespace`, and `token`. Effectively we can copy this token secret into the management cluster, and change the type to `Opaque`, and add the 'checksum' field.